packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -40,7 +89,107 @@ export class EmptyModel implements IModel {
  * }
  */
 export class ErrorModel implements IModel {
-    public readonly modelId = 'Error';
+  public readonly modelId = 'Error';
+}
+
+/**
+ * Interface share for creating or importing a model.
+ */
+export interface IModelConstruct extends cdk.IConstruct, IModel {
```

Comment:
I think `IModel` should be the `IConstruct` sub interface. I'm not sure what you mean when you talk about *simple text reference* on the comment of `IModel` itself...

Reply:
When I initially added the `IModel` interface, it was providing support just for the default Empty and Error models that are generated by default for a REST API. In that pull request, the `EmptyModel` and `ErrorModel` types were not set up to implement the `Construct` type, in the interest of keeping things simple. I can try to look into a low-impact way to converge on one IModel interface.

Reply:
I've been looking at this, and it seems weird to require the `EmptyModel` and `ErrorModel` classes to extend the full Construct class, when they are just there to be able to return "Empty" and "Error", for method responses using these built-in models.

I can, but it may just mean that I would stop exporting those types, and only expose them as static properties on the `Model` class. If that seems best, I can do it, and then converge things down to one interface.

At the moment, I've changed the naming around:
- `IModelRef` refers to the base interface that just exposes a model ID. Used for (`EmptyModel`, `ErrorModel`)
- `IModel` now refers to the full interface that inherits from `IConstruct`. Used for (`Model` and `ImportedModel`)